### PR TITLE
Explore tweaks

### PR DESCRIPTION
**Pull request by beweinreich**
Feature complete 👍 

- [x] Fix "Trends", "Daily", "Data Export" subnav to be on right hand side (hoowww???)
- [x] Tell Rob to jazz up the spaces list sidebar ... it's meh for LinkedIn and VZ
- [x] How does one make an SVG chart width 100%?
- [x] Tests, tests, tests. Because nobody likes me being a cowboy.

screen shot 2019-01-24 at 8 53 10 pm
screen shot 2019-01-24 at 8 52 54 pm
screen shot 2019-01-24 at 8 52 59 pm

### CODE REVIEW

File: src/actions/route-transition/explore.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    dispatch({ type: ROUTE_TRANSITION_EXPLORE });
+    let errorThrown;
+
+    // immediately bring them to the first space trends page if we have spaces
+    let state = getState()
+    const currentSpaces = state.spaces.data;
+    if (currentSpaces.length > 0) {
+      const spaceTree = convertSpacesToSpaceTree(currentSpaces)
+      const firstSpace = returnSpaceOrFirstChild(spaceTree[0])
+      if(firstSpace) {
```
Comment (1egoman) [inline src/actions/route-transition/explore.ts]: Can you make this `if (firstSpace) {`?

File: src/actions/route-transition/explore.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    if (!errorThrown) {
+      dispatch(collectionSpacesSet(spaces));
+    }
+
+    // if there were no current spaces, then now that we have a list, let's bring them to the first...
+    if (currentSpaces.length == 0) {
+      const spaceTree = convertSpacesToSpaceTree(spaces)
+      const firstSpace = returnSpaceOrFirstChild(spaceTree[0])
+      if(firstSpace) {
```
Comment (1egoman) [inline src/actions/route-transition/explore.ts]: `if (firstSpace) {` here too

File: src/components/explore/_styles.scss [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+.explore-sidebar-item-row.building {
+  font-weight: bold;
+}
+
+.explore-sidebar-item-row.floor {
+  font-weight: bold;
+}
+
+.explore-sidebar-item-row.space {
```
Comment (1egoman) [inline src/components/explore/_styles.scss]: Dead code

File: src/components/explore/index.tsx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+      break;
+    default:
+      page = "trends";
+      break;
+  }
+  const marginLeft = depth * 15;
+  const marginTop = (depth == 0 && spaceType != "space") ? 20 : 0;
+  const isSpace = (spaceType == "space")
+
+  var icon;
```
Comment (1egoman) [inline src/components/explore/index.tsx]: `var` => `let` 😄

File: src/components/explore/index.tsx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+import ExploreSpaceTrends from '../explore-space-trends/index';
+import ExploreSpaceDaily from '../explore-space-daily/index';
+import ExploreSpaceDataExport from '../explore-space-data-export/index';
+
+const EXPLORE_BACKGROUND = '#F5F5F7';
+const spaceFilter = filterCollection({fields: ['name']});
+
+
+function ExploreSidebarItem({selected, id, name, spaceType, depth, activePage}) {
+  var page;
```
Comment (1egoman) [inline src/components/explore/index.tsx]: `var` => `let`

File: src/helpers/convert-spaces-to-space-tree/index.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
+export default function convertSpacesToSpaceTree(spaces) {
+  let spacesByName = spaces.sort(function(a, b) {
+    return a.name.localeCompare(b.name);
+  });
+
+  var spaceTree = function(spacesToBeTreed, root) {
```
Comment (1egoman) [inline src/helpers/convert-spaces-to-space-tree/index.ts]: `var` => `let`

File: src/helpers/convert-spaces-to-space-tree/index.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    var r: any = [];
+    var o: any = {};
```
Comment (1egoman) [inline src/helpers/convert-spaces-to-space-tree/index.ts]: `var` => `let`.

Also, `r` and `o`? Can you use some more descriptive variable names?

File: src/helpers/convert-spaces-to-space-tree/index.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        o[space.parentId].children = o[space.parentId].children || [];
+        o[space.parentId].children.push(o[space.id]);
+      }
+    });
+    return r;
+  }(spacesByName, null);
+
+  // put all the spaces that aren't in a hierarchy at the bottom of the list
+  // this isn't pretty, but it works...
+  var lonelySpaces: any = []
```
Comment (1egoman) [inline src/helpers/convert-spaces-to-space-tree/index.ts]: `var` => `let`.

**Comment by beweinreich** [comment]
@bredfield found an issue when you select a time segment and go to a new page that doesnt have that time segment

File: src/reducers/spaces/index.ts [hunk 9]
```diff
@@ -146,7 +146,15 @@ export default function spaces(state=initialState, action) {
...
+    return {
+      ...state, 
+      error: null, 
+      selected: action.id, 
+      filters: {
+        ...state.filters,
+        timeSegmentGroupId: DEFAULT_TIME_SEGMENT_GROUP.id
+      }
+    };
   case ROUTE_TRANSITION_LIVE_SPACE_LIST:
```
Comment (1egoman) [inline src/reducers/spaces/index.ts]: @beweinreich I don't love your commit message but it's ... necessary?

Comment (beweinreich) [reply src/reducers/spaces/index.ts]: I’m getting rid of this. I haven’t pushed up yet, but I’m solving this in a different manner. 

The problem is moving from one space to another and the time segments differ.

Comment (1egoman) [reply src/reducers/spaces/index.ts]: Yea makes sense

**Comment by beweinreich** [comment]
@1egoman this is ready for another review.

**Comment by beweinreich** [comment]
@1egoman the main update is to the spaces reducer, where I either select the current timesegment because it exists for the selected space, or I fallback to the default timesegment if it doesn't exist for the selected space.

File: src/components/explore/README.md [hunk 10]
```diff
@@ -1,4 +1,4 @@
-# ExploreSpaceList
+# Explore
 
 This component is the root view of the explore set of pages and shows a list of all spaces. When
 clicked, the user is sent to the explore detail page for each space.
```
Comment (1egoman) [inline src/components/explore/README.md]: Might be worth updating this explanation to represent the new intention of this page.

File: src/components/explore/index.tsx [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+  if (spaces.filters.search) {
+    filteredSpaces = spaceFilter(filteredSpaces, spaces.filters.search);
+  }
+  const spacesByName = filteredSpaces.sort(function(a, b) {
+    return a.name.localeCompare(b.name);
+  });
+  const spaceTree = convertSpacesToSpaceTree(filteredSpaces)
+  const spaceList = spaces.filters.search ? spacesByName : spaceTree;
+
+  let renderExploreSidebarItem = function(space, activePage, selectedSpace, depth) {
```
Comment (1egoman) [inline src/components/explore/index.tsx]: Just like the above, can you move this `renderExploreSidebarItem` above `Explore` so it's in the parent scope? It doesn't look like it has dependencies on anything within `Explore`.

**Pull request merged**